src/server/services/redis.ts

```diff
@@ -1,26 +1,52 @@
-import redis, { RedisClient } from "redis";
+import IORedis from 'ioredis';
 import { logger } from "./logger";
 import {
   REDIS_HOST,
   REDIS_PORT,
   REDIS_PASSWORD,
   REDIS_TLS,
+  isProd,
 } from "server/config";
 
-let redisClient: RedisClient;
+export type TRedisClient = IORedis.Cluster | IORedis.Redis;
+
+let redisClient: TRedisClient | undefined;
 
 export function isRedisAvailable(): boolean {
   return REDIS_HOST !== undefined;
 }
 
-export function getRedisClient(): RedisClient {
+/**
+ * Redis client
+ * 
+ * @param cluster - Sets whether to use cluster mode. Defaults to true in production.
+ * @param clearClient - Clears the cached Redis client and initiates a new one. Useful for testing and potentially after deployment. Defaults to false.
+ */
+export function getRedisClient(cluster = isProd, clearClient = false): TRedisClient {
```

Comment:
I can't see how isProd is derived from the PR, but worth checking if the test/prod environments have these variables set correctly (in case test !isProd, and it's actually using a non-cluster without you knowing?)

Reply:
where does getRedisClient get called in the code? How would I clearClient? - wouldn't it already be "cleared" every time I restart the server?

Reply:
The only place it appears to be getting called is in the `configureExpressSession` utility (https://github.com/UKForeignOffice/lists/blob/master/src/server/components/auth/express-session.ts), which in turn gets called in the `initAuth` helper, which in turn gets called when the server initiates.

Personally I don't think the caching of the client is necessary as it appears to only be called once but I'm reluctant to bin it off as it must have been put there for a reason.

It would be cleared every time the server was restarted - in truth I've only added this flag in order to aid testing because if I instantiated it in cluster mode, when I then went to test it again in non-prod mode to check it wasn't in cluster mode it would fallback onto the cached client and the test would fail.

Reply:
`isProd` is derived from here: https://github.com/UKForeignOffice/lists/blob/master/src/server/config/server-config.ts#L14

Reply:
Looks like getRedisClient is called when initialising the express session: https://github.com/UKForeignOffice/lists/blob/1df25600410558c3d09bc159333e6b8dde69b5b0/src/server/components/auth/express-session.ts#L41.

@mogusbi-motech , I'd also be interested to understand how cluster and clearClient get set when !isProd.  Can you clarify?

Reply:
if the params are there just for testing, let's try remove them (unless it ends up taking too much time). does jest.resetModules() resolve your issue with the cached client?

Reply:
Unfortunately it doesn't

![image](https://user-images.githubusercontent.com/53015240/135826144-3172de21-0884-41d6-90cc-bc3a8e0a17a3.png)

Reply:
`jest.isolateModules` does work. Nice find @jenbutongit !